portability-transfer/src/main/java/org/datatransferproject/transfer/CallableImporter.java

```diff
@@ -63,9 +63,11 @@ public ImportResult call() throws Exception {
     boolean success = false;
     Stopwatch stopwatch = Stopwatch.createStarted();
     try {
+      idempotentImportExecutor.resetRecentErrors();
       ImportResult result = importerProvider.get()
           .importItem(jobId, idempotentImportExecutor, authData, data);
-      Collection<ErrorDetail> errors = idempotentImportExecutor.getErrors();
+
+      Collection<ErrorDetail> errors = idempotentImportExecutor.getRecentErrors();
```

Comment:
Just to clarify here, the recent errors will be per "import page" correct? What do you think of renaming the interface method to be explicit about that? ex: getLastImportPageErrors or getImportAttemptErrors?

I wonder if a better design paradigm would be to return the errors in the ImportResult instead of relying on resetting the executor? That way the executor maintains its own overall state of errors and the result would contain the page errors. Not sure if thats feasible with the current implementation, but maybe we should file a github issue to pick it up and investigate it later?

Reply:
> Just to clarify here, the recent errors will be per "import page" correct?
Yes, but I think theoretically getRecentErrors is not related with page or anything. It will contain errors until .resetRecentErrors is called. In that sense, I am reluctant to rename it to getLastImportPageErrors or getImportAttemptErrors because it may give the wrong impression that we do not need to call resetRecentErrors to clear it up.

> I wonder if a better design paradigm would be to return the errors in the ImportResult instead of relying on resetting the executor?
Yes, this will be another good design. And I did start with that to fix the issue. But unfortunately we have many importers and they don't share the same understanding of what a OK import result really means. And changing all of them to be consistent is too much effort as of now. So that's why I decided to try out this approach. Raising a github issue sounds like a good idea and will do that shortly.

Reply:
That makes sense - thanks for the explanation